CvGameCoreDLL/CvTeam.cpp

```diff
@@ -5652,6 +5652,9 @@ void CvTeam::setHasTech(TechTypes eIndex, bool bNewValue, PlayerTypes ePlayer, b
 					if (GET_PLAYER((PlayerTypes)iI).getCurrentEra() < GC.getTechInfo(eIndex).getEra())
 					{
 						GET_PLAYER((PlayerTypes)iI).setCurrentEra((EraTypes)(GC.getTechInfo(eIndex).getEra()));
+
+						// Merijn: Update yields when entering new era (for Moorish UP)
+						updateYield();

```

Comment:
I think this should happen in `setCurrentEra` and only for the Moors as part of their UP.